Tribes/src/app/app-routing.module.ts

```diff
@@ -1,17 +1,27 @@
-import { BuildingComponent } from './building/building.component';
+import { NgModule, Component } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+
+import { BuildingTypeComponent } from './building-type/building-type.component';
+import { BuildingsComponent } from './buildings/buildings.component';
 import { LeaderboardComponent } from './leaderboard/leaderboard.component';
 import { MapComponent } from './map/map.component';
 import { SettingsComponent } from './settings/settings.component';
-import { NgModule } from '@angular/core';
-import { RouterModule, Routes } from '@angular/router';
+import { MainpageComponent } from './mainpage/mainpage.component';
+import { SoldiersComponent } from './soldiers/soldiers.component';
+import { AttackComponent } from './attack/attack.component';
 
 const routes: Routes = [
   { path: '', redirectTo: '/login', pathMatch: 'full' },
+  { path: 'mainpage', component: MainpageComponent },
   { path: 'settings', component: SettingsComponent},
   { path: 'logout', redirectTo: '/', pathMatch: 'full'},
   { path: 'map', component: MapComponent},
   { path: 'leaderboard', component: LeaderboardComponent},
-  { path: 'buildings', component: BuildingComponent},
+  { path: 'buildings', component: BuildingsComponent},
+  { path: 'mainpage/soldiers', component: SoldiersComponent},
```

Comment:
we shuld do that with with child routes: something like:
` { path: 'mainpage', component: MaincComponent, children: [
      { path: 'some-other-route', component: OtherComponent },
      ......`
so the OtherCOmponents route will be: mainpage/some-other-route